superset/connectors/sqla/views.py

```diff
@@ -348,14 +353,7 @@ class TableModelView(DatasourceModelView, DeleteMixin, YamlExportMixin):  # noqa
     }
 
     def pre_add(self, table):
-        with db.session.no_autoflush:
-            table_query = db.session.query(models.SqlaTable).filter(
-                models.SqlaTable.table_name == table.table_name,
-                models.SqlaTable.schema == table.schema,
-                models.SqlaTable.database_id == table.database.id,
-            )
-            if db.session.query(table_query.exists()).scalar():
-                raise Exception(get_datasource_exist_error_msg(table.full_name))
+        models.SqlaTable.before_insert(None, None, table)
```

Comment:
Adding this fixes flashing the `General Error <class.Exception>` with the correct message on insert though overriding the `pre_update` method didn't see to remedy the issue (uncaught exception) when updating.

Reply:
Is this still an issue?

Reply:
@dpgaspar yes this still seems to be an issues, i.e., when trying to re-add the `birth_names` table,

**Without before_insert(...)**

![Screen Shot 2019-10-08 at 9 40 04 AM](https://user-images.githubusercontent.com/4567245/66415168-d5fa2580-e9af-11e9-9b91-41a2f26b12f1.png)

**With before_insert(...)**

![Screen Shot 2019-10-08 at 9 40 27 AM](https://user-images.githubusercontent.com/4567245/66415176-dabed980-e9af-11e9-9d3a-11eabb6e90f7.png)